### PR TITLE
Add prisma protocol class

### PR DESCRIPTION
**Pull request by kumaranvpl**
# Description

Please include a summary of the change and specify which issue is being addressed. Additionally, provide relevant motivation and context.

Fixes # (issue number)

## Type of change

Please delete options that are not relevant.

- [ ] Documentation (typos, code examples, or any documentation updates)
- [ ] Bug fix (a non-breaking change that resolves an issue)
- [ ] New feature (a non-breaking change that adds functionality)
- [ ] Breaking change (a fix or feature that would disrupt existing functionality)
- [ ] This change requires a documentation update

## Checklist

- [ ] My code adheres to the style guidelines of this project (`scripts/lint.sh` shows no errors)
- [ ] I have conducted a self-review of my own code
- [ ] I have made the necessary changes to the documentation
- [ ] My changes do not generate any new warnings
- [ ] I have added tests to validate the effectiveness of my fix or the functionality of my new feature
- [ ] Both new and existing unit tests pass successfully on my local environment by running `scripts/test-cov.sh`
- [ ] I have ensured that static analysis tests are passing by running `scripts/static-anaylysis.sh`
- [ ] I have included code examples to illustrate the modifications

### CODE REVIEW

File: fastagency/db/inmemory.py [hunk 1]
```diff
@@ -0,0 +1,146 @@
+import json
+from datetime import datetime
+from typing import Any, Dict, List, Optional, Union
+from uuid import UUID
+
+from fastapi import HTTPException
```
Comment (davorrunje) [inline fastagency/db/inmemory.py:6]: should not import it

File: fastagency/db/prisma.py [hunk 2]
```diff
@@ -0,0 +1,183 @@
+from contextlib import asynccontextmanager
+from datetime import datetime
+from os import environ
+from typing import Any, AsyncGenerator, Dict, List, Optional, Union
+from uuid import UUID
+
+from fastapi import HTTPException
```
Comment (davorrunje) [inline fastagency/db/prisma.py:7]: remove this: should return our Exceptions

File: fastagency/models/agents/base.py [hunk 3]
```diff
@@ -59,7 +59,9 @@ async def get_clients_from_toolboxes(self, user_id: UUID) -> List[Client]:
             if toolbox_property is None:
                 continue
 
-            toolbox_dict = await find_model_using_raw(toolbox_property.uuid)
+            toolbox_dict = await BackendDBProtocol.db().find_model(
```
Comment (davorrunje) [inline fastagency/models/agents/base.py]: why do we need find_model in agents but not in any other model?

**Review by davorrunje — CHANGES_REQUESTED** [review]

**Review by davorrunje — APPROVED** [review]

**Comment by kumaranvpl** [comment]
Fixes #9

**Pull request merged**